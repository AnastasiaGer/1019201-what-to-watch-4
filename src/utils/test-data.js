export const movieCard = {
  title: `Bohemian Rhapsody`,
  genre: `Documentary`,
  date: `2018`,
  background: `https://placeimg.com/1300/512/nature`,
  poster: `img/bohemian-rhapsody.jpg`,
  id: 178345,
  description: [
    `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
  ],
  rating: 8.1,
  scores: 800,
  director: `Bryan Singer`,
  starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  movieDurationTime: `1h 49m`,
};

export const movies = [
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: `2018`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: [
      `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
    ],
    rating: 8.2,
    scores: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    movieDurationTime: `1h 49m`,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: `2004`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/aviator.jpg`,
    id: 167456,
    description: [
      `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    ],
    rating: 7.3,
    scores: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    movieDurationTime: `1h 54m`,
  },
];

export const movieReviews = [
  {
    author: `Kate Muir`,
    id: `0`,
    date: `December 24, 2016`,
    rating: 8.9,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years`,
  },
  {
    author: `Bill Goodykoontz`,
    id: `1`,
    date: `November 18, 2015`,
    rating: 8.0,
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
  },
  {
    author: `Amanda Greever`,
    id: `2`,
    date: `November 18, 2015`,
    rating: 8.0,
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  },
  {
    author: `Matthew Lickona`,
    id: `3`,
    date: `December 20, 2016`,
    rating: 7.2,
    text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  },
  {
    author: `Paula Fleri-Soler`,
    id: `4`,
    date: `December 20, 2016`,
    rating: 7.6,
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult`,
  },
  {
    author: `Paula Fleri-Soler`,
    id: `5`,
    date: `December 20, 2016`,
    rating: 7.0,
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult`,
  },
];

export const currentMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  id: 190123,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: `8,9`,
  ratingDescription: `Very good`,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`
};
