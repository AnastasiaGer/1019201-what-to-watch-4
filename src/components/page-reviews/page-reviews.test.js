import React from 'react';
import renderer from 'react-test-renderer';
import PageReviews from './page-reviews.jsx';
import {movieReviews} from '../../utils/test-data.js';

describe(`PageReviews`, () => {
  it(`Should render correctly`, () => {
    const tree = renderer
      .create(<PageReviews
        movieReviews={movieReviews}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
