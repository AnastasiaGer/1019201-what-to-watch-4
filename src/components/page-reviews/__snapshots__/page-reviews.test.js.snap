// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PageReviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
          />
        </footer>
      </blockquote>
      <div
        className="review__rating"
      />
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
          />
        </footer>
      </blockquote>
      <div
        className="review__rating"
      />
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
          />
        </footer>
      </blockquote>
      <div
        className="review__rating"
      />
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
          />
        </footer>
      </blockquote>
      <div
        className="review__rating"
      />
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
          />
        </footer>
      </blockquote>
      <div
        className="review__rating"
      />
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
          />
        </footer>
      </blockquote>
      <div
        className="review__rating"
      />
    </div>
  </div>
</div>
`;
